indexer/src/main/java/bio/terra/tanagra/indexing/JobSequencer.java

```diff
@@ -497,14 +497,11 @@ public enum JobExecutor {
 
     public JobRunner getRunner(
         List<SequencedJobSet> jobSets, boolean isDryRun, IndexingJob.RunType runType) {
-      switch (this) {
-        case SERIAL:
-          return new SerialRunner(jobSets, isDryRun, runType);
-        case PARALLEL:
-          return new ParallelRunner(jobSets, isDryRun, runType);
-        default:
-          throw new IllegalArgumentException("Unknown JobExecution enum type: " + this);
-      }
+      return switch (this) {
```

Comment:
enhanced switch from java 17

Reply:
In general, it's not a good idea to make this sort of sweeping cleanup in a PR that is doing other, unrelated things. It makes this change harder to review because I don't know if any differences I come across are intended or not, and can cause cherry-picks, rollbacks, etc. to be more difficult. Don't bother pulling it out of this PR but something to keep in mind in the future.

Reply:
Sure -- put these changes together since all of them are cleanup / optimizations. the entire PR is low priority and we could revert the entire thing. hence put them together. dao changes were left separate because that was not a cleanup, but refactored (for later use in next pr)